File: BussinessObjects/Models/Notification.cs
namespace BussinessObjects.Models;

public class Notification
{
    public long NotificationId { get; set; }

    public int? RecipientUserId { get; set; }

    public int? RecipientHelperId { get; set; }

    public string Title { get; set; } = null!;

    public string Message { get; set; } = null!;

    public string? NotificationType { get; set; }

    public string? ReferenceId { get; set; }

    public bool? IsRead { get; set; }

    public DateTime? ReadTime { get; set; }

    public DateTime? CreationTime { get; set; }

    public DateTime? SentTime { get; set; }

    public virtual Helper? RecipientHelper { get; set; }

    public virtual User? RecipientUser { get; set; }
}

File: src/types/notification.ts
// Notification Types
export interface Notification {
  notificationId: number;
  recipientUserId: number | undefined;
  recipientHelperId: number | undefined;
  title: string;
  message: string;
  notificationType: NotificationType | string;
  referenceId: string | null;
  isRead: boolean;
  readTime: string | null;
  creationTime: string;
  sentTime: string;
}

export type NotificationType =
  | 'AccountRestoration'
  | 'AccountSuspension'
  | 'BookingCancelled'
  | 'BookingUpdate'
  | 'NewBooking'
  | 'PasswordChange'
  | 'ProfileUpdate'
  | 'ApplicationStatus';

export interface NotificationResponse {
  success: boolean;
  statusCode: number;
  message: string;
  data: Notification[];
  timestamp: string;
  requestId: string;
}

export interface NotificationCountResponse {
  success: boolean;
  statusCode: number;
  message: string;
  data: number;
  timestamp: string;
  requestId: string;
}

export interface MarkReadResponse {
  success: boolean;
  statusCode: number;
  message: string;
  data: string;
  timestamp: string;
  requestId: string;
}

export interface NotificationState {
  notifications: Notification[];
  unreadCount: number;
  loading: boolean;
  error: string | null;
} 